### PR TITLE
Map remote views

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/planner.js [hunk 1]
```diff
@@ -107,44 +114,67 @@ class AssignViewsByTagAndType extends Strategy {
...
+        }
 
-        var responses = arc.findViews(type, tags).map(newView =>
+        if (tags.length > 0) {
+          // score bump for matching tag information
+          score += 4;
+          var views = arc.findViews(type, {tag: tags[0]});
+          views = views.concat(mappable.map(arc => arc.findViews(type, {tag: tags[0]})).reduce((a,b) => a.concat(b), []));
+        }
+        else
```
Comment (dstockwell) [inline runtime/planner.js]: formatting

File: runtime/recipe/type-checker.js [hunk 2]
```diff
@@ -20,8 +20,10 @@ class TypeChecker {
     for (var i = 1; i < list.length; i++) {
       let result = TypeChecker.compareTypes(baseType, list[i], variableResolutions);
       baseType = result.type;
-      if (!result.valid)
+      if (!result.valid) {
+        debugger;
```
Comment (dstockwell) [inline runtime/recipe/type-checker.js]: oops

File: runtime/recipe/walker-base.js [hunk 3]
```diff
@@ -49,6 +49,18 @@ class WalkerBase extends Strategizer.Walker {
 
             newRecipes.push({recipe: newRecipe, score});
           }
+          /*
+          if (newRecipes.length > 10) {
```
Comment (dstockwell) [inline runtime/recipe/walker-base.js]: do we want this?

File: runtime/planner.js (hunk 1 above)
Comment (shans) [inline runtime/planner.js]: Done.

File: runtime/recipe/type-checker.js (hunk 2 above)
Comment (shans) [inline runtime/recipe/type-checker.js]: Removed.

File: runtime/recipe/walker-base.js (hunk 3 above)
Comment (shans) [inline runtime/recipe/walker-base.js]: Nah

**Pull request merged**